instrumentation/src/java/org/oppia/android/instrumentation/TestNetworkModule.kt

```diff
@@ -0,0 +1,84 @@
+package org.oppia.android.instrumentation
+
+import dagger.Module
+import dagger.Provides
+import okhttp3.OkHttpClient
+import org.oppia.android.data.backends.gae.JsonPrefixNetworkInterceptor
+import org.oppia.android.data.backends.gae.NetworkApiKey
+import org.oppia.android.data.backends.gae.OppiaRetrofit
+import org.oppia.android.data.backends.gae.RemoteAuthNetworkInterceptor
+import org.oppia.android.data.backends.gae.api.ClassroomService
+import org.oppia.android.data.backends.gae.api.FeedbackReportingService
+import org.oppia.android.data.backends.gae.api.TopicService
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import javax.inject.Singleton
+
+/**
+ * Module which provides all required dependencies about network
+ *
+ * Sample resource: https://github.com/gahfy/Feed-Me/tree/unitTests
+ */
+@Module
+class TestNetworkModule {
```

Comment:
Here & elsewhere: I'm a bit concerned with calling everything 'test' because it may be conflated with other tests that shouldn't be using these things (e.g. robolectric unit tests). Suggest instead using 'EndToEndTest' for each place where 'Test' is used currently for these new components.

Reply:
Done.
Not sure if you meant to include OppiaTestApplication and TestApplicationComponent too here.